src/Common/src/Interop/Windows/SspiCli/Interop.SSPI.cs

```diff
@@ -25,7 +26,13 @@ internal struct CredHandle
 
             public bool IsZero
             {
-                get { return dwLower == IntPtr.Zero && dwUpper == IntPtr.Zero; }
+                [MethodImpl(MethodImplOptions.AggressiveInlining)]
+                get
+                {
+                    // Ternary operator returning true/false prevents redundant asm generation:
+                    // https://github.com/dotnet/coreclr/issues/914
+                    return dwLower == IntPtr.Zero && dwUpper == IntPtr.Zero ? true : false;
```

Comment:
This change seems like a temporary workaround to the CLR or JIT. It is not obvious at all to most developers that they need to use a ternary operator. In fact, we have had many PRs over the years where code like this change above has been changed to the original, shortened form `return dwLower == IntPtr.Zero && dwUpper == IntPtr.Zero;`

I don't see the benefit of this change other than an unproven small micro-perf benefit. I would rather see the CLR and related compiler/JIT fixed to make the original pattern performant. This change further reduces readability.

Reply:
> This change seems like a temporary workaround to the CLR or JIT.

4.25 years, but I'm not counting https://github.com/dotnet/coreclr/issues/914 😉 

> I don't see the benefit of this change other than an unproven small micro-perf benefit.

It drops 10 asm instructions

```diff  
-   setz al               
-   movzx eax, al         
-   jmp L0039             
-   xor eax, eax          
-   jmp L0040             
-   mov eax, 0x1          
-   test eax, eax         
-   setz al               
-   movzx eax, al         
-   test eax, eax   
```

>  I would rather see the CLR and related compiler/JIT fixed to make the original pattern performant.

Me too; will back out and raise more general issue.

Reply:
https://github.com/dotnet/coreclr/issues/914#issuecomment-521083983